bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/item/EnrichedItemDTOMapper.java

```diff
@@ -51,6 +56,10 @@ public class EnrichedItemDTOMapper {
 
     private static final Pattern EXTRACT_TRANSFORM_FUNCTION_PATTERN = Pattern.compile("(.*?)\\((.*)\\):(.*)");
 
+    private static final String DATE_FORMAT_PATTERN_WITH_TZ_RFC = "yyyy-MM-dd'T'HH:mm[:ss[.SSSSSSSSS]]Z";
+    private static final DateTimeFormatter FORMATTER_TZ_RFC = DateTimeFormatter
```

Comment:
Is this pattern identical to DateTimeFormatter.ISO_OFFSET_DATE_TIME? In any case the .SSSSSSSSS is incorrect, should be .nnnnnnnnn in my mind.

Reply:
I have to admit this was duplicated from `DateTimeType` itself, thinking I would later remove it there - which didn't happen. I'll see if it can be refactored while maintaining backwards compatiblity, and also get rid of the duplication.

Reply:
In `DateTimeType` I have now exposed a method overload of `toFullString` taking a `ZoneId` - and added full test coverage for this. As a consequence, this duplicated method and `DateTimeFormatter` was removed.

For now, I have left the method and `DateTimeFormatter` as it was before this PR, but we now have test coverage to make sure refactoring will not break the backwards compatibility.

Reply:
It seems `DateTimeFormatter.ISO_OFFSET_DATE_TIME` does not provide the same output:
```
[ERROR] Failures: 
[ERROR]   DateTimeTypeTest.createDate:360 expected: <2014-03-30T10:58:47.033+0000> but was: <2014-03-30T10:58:47.033Z>
[ERROR]   DateTimeTypeTest.createDate:360 expected: <2014-03-30T12:58:47.033+0200> but was: <2014-03-30T12:58:47.033+02:00>
[ERROR]   DateTimeTypeTest.createDate:360 expected: <2014-03-30T12:58:47.033+0200> but was: <2014-03-30T12:58:47.033+02:00>
[ERROR]   DateTimeTypeTest.createDate:360 expected: <2014-03-30T05:58:47.033-0200> but was: <2014-03-30T05:58:47.033-02:00>
[ERROR]   DateTimeTypeTest.createDate:360 expected: <2014-03-30T12:58:47.033-0200> but was: <2014-03-30T12:58:47.033-02:00>
[ERROR]   DateTimeTypeTest.createDate:360 expected: <1970-01-01T10:58:47.000+0000> but was: <1970-01-01T10:58:47Z>
[ERROR]   DateTimeTypeTest.createDate:360 expected: <1970-01-01T10:58:00.000+0000> but was: <1970-01-01T10:58:00Z>
[ERROR]   DateTimeTypeTest.createDate:360 expected: <1970-01-01T10:58:47.000+0100> but was: <1970-01-01T10:58:47+01:00>
[ERROR]   DateTimeTypeTest.createDate:360 expected: <1970-01-01T10:58:00.000+0100> but was: <1970-01-01T10:58:00+01:00>
[ERROR]   DateTimeTypeTest.createDate:360 expected: <2014-03-30T00:00:00.000+0000> but was: <2014-03-30T00:00:00Z>
[ERROR]   DateTimeTypeTest.createDate:360 expected: <2014-03-30T10:58:47.033+0000> but was: <2014-03-30T10:58:47.033Z>
[ERROR]   DateTimeTypeTest.createDate:360 expected: <2014-03-30T08:58:47.033+0000> but was: <2014-03-30T08:58:47.033Z>
[ERROR]   DateTimeTypeTest.createDate:360 expected: <2014-03-30T10:58:47.230+0000> but was: <2014-03-30T10:58:47.23Z>
[ERROR]   DateTimeTypeTest.createDate:360 expected: <2014-03-30T10:58:47.000+0000> but was: <2014-03-30T10:58:47Z>
[ERROR]   DateTimeTypeTest.createDate:360 expected: <2014-03-30T12:58:47.033+0200> but was: <2014-03-30T12:58:47.033+02:00>
[ERROR]   DateTimeTypeTest.createDate:360 expected: <2014-03-30T10:58:47.033+0200> but was: <2014-03-30T10:58:47.033+02:00>
[ERROR]   DateTimeTypeTest.createDate:360 expected: <2014-03-30T10:58:47.000+0200> but was: <2014-03-30T10:58:47+02:00>
[ERROR]   DateTimeTypeTest.createDate:360 expected: <2014-03-30T15:58:47.000+0500> but was: <2014-03-30T15:58:47+05:00>
[ERROR]   DateTimeTypeTest.createDate:360 expected: <2014-03-30T10:58:47.000+0200> but was: <2014-03-30T10:58:47+02:00>
[ERROR]   DateTimeTypeTest.toFullStringWithZone:412 
Expected: is "2024-11-11T20:39:00.000+0000"
     but: was "2024-11-11T20:39:00Z"
[ERROR]   DateTimeTypeTest.toFullStringWithZone:412 
Expected: is "2024-11-11T20:39:00.000+0000"
     but: was "2024-11-11T20:39:00Z"
[ERROR]   DateTimeTypeTest.toFullStringWithZone:412 
Expected: is "2024-11-11T20:39:00.000000001+0000"
     but: was "2024-11-11T20:39:00.000000001Z"
[ERROR]   DateTimeTypeTest.toFullStringWithZone:412 
Expected: is "2024-11-11T20:39:00.123+0000"
     but: was "2024-11-11T20:39:00.123Z"
[ERROR]   DateTimeTypeTest.toFullStringWithZone:412 
Expected: is "2024-11-11T20:39:00.123456+0000"
     but: was "2024-11-11T20:39:00.123456Z"
[ERROR]   DateTimeTypeTest.toFullStringWithZone:412 
Expected: is "2024-11-11T20:39:00.123456789+0000"
     but: was "2024-11-11T20:39:00.123456789Z"
[ERROR]   DateTimeTypeTest.toFullStringWithZone:412 
Expected: is "2024-11-11T21:39:00.123+0100"
     but: was "2024-11-11T21:39:00.123+01:00"
[ERROR]   DateTimeTypeTest.toFullStringWithZone:412 
Expected: is "2024-11-10T23:59:59.999-0500"
     but: was "2024-11-10T23:59:59.999-05:00"
```

Reply:
The parser test does accept the format above as valid input, so why should this be invalid output?

Reply:
With da2b50a2e71da3152248b349a4fc77a11db81ead this duplicated method was removed, so what is left is the original implementation of `toString()`/`toFullString()`. This is unchanged by this PR to remain backwards compatible. The `DateTimeType` String constructor supports many different inputs, but `toFullString()` can only provide one specific output.